sdk/core/core-amqp/src/auth/iotSas.ts

```diff
@@ -1,25 +1,25 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import { SasTokenProvider } from "./sas";
-import { TokenInfo } from "./token";
+import { SharedKeyCredential } from "./sas";
+import { AccessToken } from "@azure/core-http";
 import { Buffer } from "buffer";
 
 /**
- * @class IotSasTokenProvider
+ * @class IotSharedKeyCredential
  * @ignore
- * Defines the SasTokenProvider for IotHub.
+ * Defines the IotSharedKeyCredential for IotHub.
  */
-export class IotSasTokenProvider extends SasTokenProvider {
+export class IotSharedKeyCredential extends SharedKeyCredential {
   /**
    * Gets the sas token for the specified audience for IotHub.
    * @ignore
    * @param {string} [audience] - The audience for which the token is desired. If not
    * provided then the Endpoint from the connection string will be applied.
    */
-  async getToken(audience?: string): Promise<TokenInfo> {
+  async getToken(audience?: string): Promise<AccessToken> {
```

Comment:
Why is this an async function? There is nothing async happening inside :)

Reply:
It extends `SharedKeyCredential` and the getMethod() of `SharedKeyCredential` is async and returns a promise. I wanted to make things consistent with aad getToken().

Even though if I remove the async from the `SharedKeyCredential` getToken() method, here I have to add await
![image](https://user-images.githubusercontent.com/45216704/59558828-289e7480-8fb0-11e9-8de5-b6a85560732f.png)

because the type of tokenCredential in connectionContext is `SharedKeyCredential | TokenCredential` and TokenCredential  getToken() method returns a promise.

Reply:
Even in `SharedKeyCredential`, the `getToken()` doesn't need to be async. There is no async work going on there as well.

I am guessing the code snippet you have shared is from the samples.
Since `audience` is being passed, it should be well known to the consumer of the code that the `connectionContext.tokenCredential` here is a SharedKeyCredential and therefore should be able to cast it appropriately

```
const sharedTokenCredential = <SharedKeyCredential>connectionContext.tokenCredential
const tokenObject = sharedTokenCredential.getToken(audience);
```